tree-sitter/src/TreeSitter/Unmarshal.hs

```diff
@@ -385,3 +389,17 @@ instance (Unmarshal t, Selector c) => GUnmarshalProduct (M1 S c (Rec1 t)) where
       _   -> liftIO . throwIO . UnmarshalError $ "type '" <> datatypeName <> "' expected a node but got multiple"
     where
     fieldName = selName @c undefined
+
+
+class GHasAnn a t where
+  gann :: t a -> a
+
+instance GHasAnn a f => GHasAnn a (M1 i c f) where
+  gann = gann . unM1
+
+instance (GHasAnn a l, GHasAnn a r) => GHasAnn a (l :+: r) where
+  gann (L1 l) = gann l
+  gann (R1 r) = gann r
+
+instance {-# OVERLAPPABLE #-} HasField "ann" (t a) a => GHasAnn a t where
+  gann = getField @"ann"
```

Comment:
I was kind of surprised that this worked without trouble since it clearly overlaps with the other two instances, but it seems to work just fine.